cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1316,12 +1335,31 @@ private class ProposalExecutionRunnable implements Runnable {
       _replicationThrottle = replicationThrottle;
       _isTriggeredByUserRequest = isTriggeredByUserRequest;
       _lastSlowTaskReportingTimeMs = -1L;
+      if (removedBrokers != null && !removedBrokers.isEmpty()) {
+        _executionTimerInvolveBrokerRemovalOrDemotion = _proposalExecutionTimerInvolveBrokerRemoval;
+      } else if (demotedBrokers != null && !demotedBrokers.isEmpty()) {
+        _executionTimerInvolveBrokerRemovalOrDemotion = _proposalExecutionTimerInvolveBrokerDemotionOnly;
+      } else {
+        _executionTimerInvolveBrokerRemovalOrDemotion = null;
+      }
     }
 
     public void run() {
       LOG.info("Starting executing balancing proposals.");
-      UserTaskManager.UserTaskInfo userTaskInfo = initExecution();
-      execute(userTaskInfo);
+      final Timer.Context ctxExecutionBrokerRemovalOrDemotion = _executionTimerInvolveBrokerRemovalOrDemotion != null
```

Comment:
Do you actually want a timer or just a gauge? 

I think a timer helps to get statistical results like avg, p99, etc, especially when there are many concurrent operations. In this case, since there is only one execution at a time, I think a gauge might be more helpful. What do you think

Reply:
Ok that a fair point.